frontend/app/controllers/TierController.scala

```diff
@@ -199,7 +199,7 @@ object TierController extends Controller with ActivityTracking
       }
 
       def doUpgrade(): Future[Result] = {
-        val upgrade = memberService.upgradePaidSubscription(paidMember, target, form, CampaignCode.fromRequest)
+        val upgrade = memberService.upgradePaidSubscription(paidMember, target, form, CampaignCode.fromRequest, RefererUrl.fromRequest, RefererPageviewId.fromRequest)
```

Comment:
I'd like to avoid adding multiple parameters that have to be handed down in all call invocations - could these fields be encapsulated in a single content case class? Should probably have all 3 of these in it:

* CampaignCode.fromRequest
* RefererUrl.fromRequest
* RefererPageviewId.fromRequest